workloads/kube-burner/common.sh

```diff
@@ -242,3 +246,70 @@ prep_networkpolicy_workload() {
   oc apply -f workloads/networkpolicy/clusterrole.yml
   oc apply -f workloads/networkpolicy/clusterrolebinding.yml
 }
+
+churn() {
+  log "Starting to churn workload"
+
+  churn_start=`date +%s`
+  churn_end_time=$((${churn_start} + ${CHURN_DURATION}*60))
+
+  log "Churn duration: ${CHURN_DURATION} minutes"
+  log "Churn wait duration: ${CHURN_WAIT} seconds"
+  log "Churn percentage: ${CHURN_PERCENT}%"
+  log "Churn type: ${CHURN_TYPE}"
+
+  namespace_array=(`kubectl get namespaces -l kube-burner-uuid=${UUID} --no-headers | awk '{print $1}'`)
+  namespace_count=${#namespace_array[@]}
+  
+  # The number of iterations we need to modify per round of churn (min 1)
+  modify_count=$((JOB_ITERATIONS*CHURN_PERCENT/100))
+  if [[ ${modify_count} -eq 0 ]]; then modify_count=1; fi
+  
+  current_time=`date +%s` 
+  while [ ${current_time} -le ${churn_end_time} ]; do
+    for ((i=0; i<${modify_count}; i++)); do
+      #Pick random Namespace from the list
+      rand_ns=$(($RANDOM%${namespace_count}))
+
+      if [[ ${CHURN_TYPE} == "pod" ]]; then
+        rep_sets=(`kubectl get replicasets -n ${namespace_array[$rand_ns]} --no-headers | awk '{print $1}'`)
+        #Pick a random replicaset from the namespace
+        rand_rep=$(($RANDOM%${#rep_sets[@]}))
+        #set to 0 replicas; it will automatically get swapped back to the appropriate number of replicas
+        kubectl scale --replicas 0 replicaset ${rep_sets[$rand_rep]} -n ${namespace_array[$rand_ns]} > /dev/null &
+      elif [[ ${CHURN_TYPE} == "namespace" ]]; then
+	#Churn type is namespace
+	#Get all relevent configs
+	kubectl get -o yaml namespace ${namespace_array[$rand_ns]} > churn_namespace.yaml
+	kubectl get -o yaml configmaps -l kube-burner-uuid=${UUID} -n ${namespace_array[$rand_ns]} > churn_configmaps.yaml
+	kubectl get -o yaml secrets -l kube-burner-uuid=${UUID} -n ${namespace_array[$rand_ns]} > churn_secrets.yaml
+	kubectl get -o yaml all -l kube-burner-uuid=${UUID} -n ${namespace_array[$rand_ns]} > churn_all.yaml
```

Comment:
Why do we need get all?, you can do the same with just one command

`kubectl get -o yaml cm,secrets,deploy,route -l kube-burner-uuid=${UUID}`

Reply:
I found when writing them all to the same yaml file it had issues re-deploying things as expected. Breaking into smaller groups worked every time.